src/brand/bhyve/init

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python3.5
+
+# {{{ CDDL HEADER
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# }}}
+
+# Copyright 2020 OmniOS Community Edition (OmniOSce) Association.
+
+import logging, os, subprocess, sys, time, re, getopt
+import xml.etree.ElementTree as etree
+from pprint import pprint, pformat
+
+zonexml = '/etc/zone.xml'
+uuidfile = '/etc/uuid'
+testmode = False
+
+try:
+    opts, args = getopt.getopt(sys.argv[1:], "tx:u:")
+except getopt.GetoptError:
+    print("init [-t] [-x <xml file>] [-u <uuid file>]")
+    sys.exit(2)
+for opt, arg in opts:
+    if opt == '-t':
+        testmode = True
+    elif opt == '-x':
+        zonexml = arg
+    elif opt == '-u':
+        uuidfile = arg
+
+if testmode:
+    logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+else:
+    logging.basicConfig(filename='/tmp/init.log', filemode='a',
+        level=logging.DEBUG)
+
+# Default values
+opts = {
+    'acpi':         'on',
+    'vcpus':        '1',
+    'ram':          '1G',
+    'bootrom':      'BHYVE_RELEASE_CSM',
```

Comment:
What's the difference between CSM and not-CSM ROMs? 
Debian ISO evidently don't show any signs of activity with CSM ROM,

Reply:
@Toasterson @citrus-it
Do I understand correctly that vnc console is supported only with UEFI firmware (and, evidently, only for UEFI guests)?
If it's true, does it make sense to use csd firmware by default?

Reply:
CSM is the a shorthand for what is commonly known as traditional BIOS. the non CSM Firmwares are UEFI Firmware. VNC should be supported with CSM aswell. It may be that we are out of date though. The UEFI Firmware does not support BIOS formatted disks and thus some systems may need the BIOS Firmware to be ported over otherwise they need to be re-installed. But we could default to the UEFI one if we aren't already. @citrus-it What is your Opinion on this?

Reply:
legacy (CSM) boot does not support the VNC console.
The CSM firmware variant should be able to boot either UEFI (with VNC) or legacy (without VNC) which is why it's the default in OmniOS.

However, sometimes you get better results by switching to the UEFI ROM.

Reply:
Does the VNC never come up for the OS or is it only the boot output which cannot be managed via VNC?

Reply:
For me non-CSM ROMs seem to be more adequate. Somehow with CSM I even don't see console  (with zlogin -C) during FreeBSD 12 installation.

Reply:
That may be dependant on OS. For Archlinux I had to configure the OS to use the serial port. Does VNC work when booting with CMS for the OS. I would also prefer the UEFI ROm as default.

Reply:
Then set ROMs to non-CSM by default, update documentation accordingly and let's megre it.

Reply:
Done